python_modules/dagit/dagit/webserver.py

```diff
@@ -74,6 +74,16 @@ def make_request_context(self, conn: HTTPConnection) -> BaseWorkspaceRequestCont
     def build_middleware(self) -> List[Middleware]:
         return [Middleware(DagsterTracedCounterMiddleware)]
 
+    def make_security_headers(self) -> dict:
+        return {
+            "Cache-Control": "no-store",
```

Comment:
Cache-Control no-store is a slight behavior change here (beyond just a security change). I think it's good not to cache index.html so that updates to the react app are propagated immediately, but I think caching with must-revalidate might be a bit better so that it can 304 Not Modified rather than having to fetch all the data every time?

Reply:
Given that we populate the CSP nonce into the HTML per-request, I think we never want a cached document. 🤔 What do you think?